wordpress-mobile/WordPress-Android
# Api26/jobschedule token refresh

This PR uses `JobIntentService` for GCM token refreshes, as specified in https://developers.google.com/cloud-messaging/android/client

```If you're targeting Android 8.0 (API level 26) or higher in your app development, use Firebase Cloud Messaging (FCM). Otherwise, use the JobIntentService class instead of the IntentService to handle token refreshes.```

This is needed given that when a new token is assigned to the device/app pair, we need to inform our servers (and Helpshift) by using a network connection. This could exceed the 10-second background processing limit given to background Services since Android O, hence the need to use a `JobIntentService` for this case.

`JobIntentService` should work exactly the same for pre-Android O devices.

This is the first of a series of PRs that will tackle each of our background services and wrap them up in `JobIntentService` as we see fit. I'll use this PR as a model for future PRs along these lines.


To test:
*CASE A: token sending*
1. place a breakpoint on the first line in `onHandleWork()` method of `GCMRegistrationIntentService` class
2. run the app
3. Observe the app stops at the breakpoint.

Test both pre and post Android O platforms.

*CASE B: token refresh*
Note: The only way to force a token refresh is on a fresh install. I had trouble cleaning the app 100% from my device, and could only do so by clearing data, stopping it, uninstalling it, and finally restarting my phone before installing it again.
1. To test a fresh install, place the breakpoint in the `onTokenRefresh()` method in the `InstanceIDService` service class
2. after installing, debug the app. It should stop at the breakpoint.

cc @nbradbury
